SU2_CFD/include/iteration/modules/CSolverModule.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include "../../Common/include/toolboxes/CModule.hpp"
+
+class CGeometry;
+class CConfig;
+class CSolver;
+class COutput;
+
+class CSolverModule : public CModule<CSolverModule>{
+
+public:
+  CSolverModule(bool enabled_ = true) : CModule(enabled_) {};
+
+  virtual void PreIterationHook(COutput *output, CConfig* config, CGeometry* geometry, CSolver** solver) {};
```

Comment:
Its not directly necessary here to define the functions virtual, but I makes the interface more clear.